lib/active_record_shards/migration.rb

```diff
@@ -1,25 +1,31 @@
 # frozen_string_literal: true
 module ActiveRecord
   class Migrator
-    class << self
-      [:up, :down, :run].each do |m|
-        define_method("#{m}_with_sharding") do |*args|
-          ActiveRecord::Base.on_shard(nil) do
-            send("#{m}_without_sharding", *args)
-          end
-          ActiveRecord::Base.on_all_shards do
-            send("#{m}_without_sharding", *args)
-          end
+    [:run, :migrate].each do |method|
+      define_method("#{method}_with_sharding") do |*args|
+        ActiveRecord::Base.on_shard(nil) do
+          send("#{method}_without_sharding", *args)
+        end
+        ActiveRecord::Base.on_all_shards do
+          send("#{method}_without_sharding", *args)
         end
-        alias_method :"#{m}_without_sharding", m.to_sym
-        alias_method m.to_sym, :"#{m}_with_sharding"
       end
+      alias_method :"#{method}_without_sharding", method.to_sym
+      alias_method method.to_sym, :"#{method}_with_sharding"
```

Comment:
This patch (lines 4–16) could be extracted into a separate commit – or a separate pull request, even.

I am thinking, perhaps it would be better not to use metaprogramming when we are only overriding 2 methods. E.g.:

```rb
    def run_with_sharding(*args)
      ActiveRecord::Base.on_shard(nil) { run_without_sharding(*args) }
      ActiveRecord::Base.on_all_shards { run_without_sharding(*args) }
    end
    alias_method :run_without_sharding, :run
    alias_method :run, :run_with_sharding

    def migrate_with_sharding(direction)
      ActiveRecord::Base.on_shard(nil) { migrate_without_sharding(direction) }
      ActiveRecord::Base.on_all_shards { migrate_without_sharding(direction) }
    end
    alias_method :migrate_without_sharding, :migrate
    alias_method :migrate, :migrate_with_sharding
```